storybook/src/stories/searchable-dropdown/component.helpers.tsx

```diff
@@ -2,123 +2,67 @@ import {
+    return Object.assign(
       {
-        id: '8aa0d62f-21d4-4933-a1e1-823a8de7c14d',
-        title: 'Project4 prod',
-        subTitle: 'A Fusion context',
-        graphic: 'settings',
+        id: faker.string.uuid(),
+        title: faker.company.name(),
+        subTitle: faker.company.catchPhrase(),
       },
-    ],
-  },
-  { id: '8aa0d62f-21d4-4933-a1e1-823a8de7c15a', title: 'Unsectioned prod', subTitle: 'Some random item' },
-  { id: '8aa0d62f-21d4-4933-a1e1-823a8de7c15b', title: 'Unsectioned2 prod', subTitle: 'Some random item' },
-  { id: '8aa0d62f-21d4-4933-a1e1-823a8de7c15c', title: 'Unsectioned3 prod', subTitle: 'Some random item' },
-  { id: '8aa0d62f-21d4-4933-a1e1-823a8de7c15d', title: 'Unsectioned4 prod', subTitle: 'Some random item' },
-];
+      merge ?? {},
+    );
+  });
+};
+
+/**
+ * Faker data generator for section with children.
+ */
+const getSections = (len: number, kids: number) => {
+  return new Array(len).fill(undefined).map((_, i) => {
+    const opts: Partial<SearchableDropdownResultItem> = { graphic: 'list' };
+    if ((i + 1) % 2 === 0) {
+      opts.graphic = orgIcon;
+      opts.graphicType = 'inline-svg';
+    } else if ((i + 1) % 3 === 0) {
+      opts.graphic = eqOneIcon;
+      opts.graphicType = 'inline-svg';
+    } else if ((i + 1) % 4 === 0) {
+      opts.graphic = reviewIcon;
+      opts.graphicType = 'inline-svg';
+    }
 
-/* Dummy api handler */
+    return {
+      id: faker.string.uuid(),
+      title: `Section ${faker.company.buzzVerb()}`,
+      type: 'section',
+      children: getItems(kids, opts),
+    } satisfies SearchableDropdownResultItem;
+  });
+};
+
+/**
+ * Example api handler
+ * Takes the query string to search for and return matching example projects.
+ * Query string min length to start "http request" is 2 chars.
+ */
 const apiItems = (query: string): SearchableDropdownResult => {
   /* min length of query string */
   const min = 2;
-  const matched = [];
-  if (!query || query.length < min) {
-    matched.push(singleItem({ title: `Need ${min - query.length} more chars`, isDisabled: true }));
-    return matched;
+  if (!query || query.length < 2) {
+    return [{ id: faker.string.uuid(), title: `Need ${min - query.length} more chars`, isDisabled: true }];
   }
 
-  /* Recursive func for matching in children  */
-  for (const item of allItems as SearchableDropdownResult) {
-    const entry = { ...item };
-    // Match against children in sections
-    if (entry.type === 'section' && entry.children?.length) {
-      entry.children = entry.children.filter((i) => i.title && i.title.toLowerCase().indexOf(query) > -1);
-      if (entry.children.length) {
-        matched.push(entry);
-      }
-    } else if (entry.title && entry.title.toLowerCase().indexOf(query) > -1) {
-      matched.push(entry);
-    }
-  }
-
-  if (!matched.length) {
-    matched.push(singleItem({ title: 'No matches...', isDisabled: true }));
-  }
-
-  return matched;
```

Comment:
For the code example it might be a good idea to include a no-match result.

Reply:
api now returns errors when no match and to short query string.